raiden/tests/integration/test_matrix_transport.py

```diff
@@ -306,6 +312,70 @@ def test_matrix_message_sync(
     transport1.get()
 
 
+@pytest.mark.parametrize('number_of_nodes', [2])
+@pytest.mark.parametrize('channels_per_node', [1])
+@pytest.mark.parametrize('number_of_tokens', [1])
+def test_matrix_tx_error_handling(
+    skip_if_not_matrix,
+    raiden_chain,
+    token_addresses,
+):
+    """ Test exception propagation for transactions triggered by the transport greenlet.
+    """
+    app0, app1 = raiden_chain
+    token_network = views.get_token_network_by_token_address(
+        views.state_from_app(app0),
+        app0.raiden.default_registry.address,
+        token_addresses[0],
+    )
+    channel_identifier = token_network.partneraddresses_to_channelidentifiers[app1.raiden.address]
+    assert len(channel_identifier) == 1
+    channel_identifier = channel_identifier[0]
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        views.state_from_app(app0),
+        app0.raiden.default_registry.address,
+        token_addresses[0],
+    )
+    burn_eth(app0.raiden)
+
+    def make_tx(*args, **kwargs):
+        """ We forego the transport itself and push right to RaidenService.handle_state_change.
+        This is meant to be used as a transport callback."""
+        close_channel = ActionChannelClose(
+            token_network_identifier=token_network_identifier,
+            channel_identifier=channel_identifier,
+        )
+        app0.raiden.handle_state_change(close_channel)
+
+    app0.raiden.transport._client.add_presence_listener(make_tx)
+    waiting.wait_for_events_in_wal(
+        raiden=app0.raiden,
+        event_type=ContractSendChannelClose,
+        count=1,
+        retry_timeout=1.0,
+    )
+    # assert the `close` did fail
+    channelstate = views.get_channelstate_by_token_network_and_partner(
+        chain_state=views.state_from_raiden(app0.raiden),
+        token_network_id=token_network_identifier,
+        partner_address=app1.raiden.address,
+    )
+    assert channelstate.close_transaction.started_block_number is not None
+    max_wait = (
+        channelstate.close_transaction.started_block_number +
+        (2 * DEFAULT_NUMBER_OF_BLOCK_CONFIRMATIONS)
+    )
+    waiting.wait_for_block(
+        raiden=app0.raiden,
+        block_number=max_wait,
+        retry_timeout=.5,
+    )
+    # FIXME: TransactionExecutionStatus is not yet set to FAILURE, so asserting no SUCCESS
+    assert channelstate.close_transaction.result != TransactionExecutionStatus.SUCCESS
+    # assert transport is still healthy
+    assert not app0.raiden.transport._stop_event.ready()
```

Comment:
@konradkonrad why the transport should /not/ be stopped on a unrecoverable exception?